.github/workflows/_shared-build.yaml

```
@@ -94,8 +94,8 @@ jobs:
                   ref: ${{ inputs.ref }}
                   repository: ${{ inputs.repository }}
                   submodules: recursive
-                  # Safe: this job is contents:read and this workflow takes no secrets input.
-                  allow-unsafe-pr-checkout: true
+                  persist-credentials: false
+                  allow-unsafe-pr-checkout: ${{ github.event_name == 'pull_request_target' }}
```

actionlint (1.7.12)

[error] input "allow-unsafe-pr-checkout" is not defined in action "actions/checkout@v6". available inputs are "clean", "fetch-depth", "fetch-tags", "filter", "github-server-url", "lfs", "path", "persist-credentials", "ref", "repository", "set-safe-directory", "show-progress", "sparse-checkout", "sparse-checkout-cone-mode", "ssh-key", "ssh-known-hosts", "ssh-strict", "ssh-user", "submodules", "token"

(action)